### ISSUE TITLE
Noclip locomotion name cannot be customized

### ISSUE THREAD

**Issue by amplified1** [Yellow-Dog-Man/Resonite-Issues#2618]
### Is your feature request related to a problem? Please describe.

The name of Noclip Locomotion is hard-coded and cannot be changed, this makes working with the locomotion more difficult and makes it impossible to use the switch locomotion module node to change to a custom variant of noclip.

### Describe the solution you'd like

Have the Noclip locomotion module pull from its slot name like the expected behavior of other locomotion modules.

### Describe alternatives you've considered

Add options to physical locomotion, being the most customizable locomotion module currently in the game, to allow it to replicate the behaviors of No-clip, although I don't know if this is an ideal solution, as it is kind-of an oxymoron.

### Requesters

`@_amplified_ @euphieeuphoria @alizardguy`

**Comment by 3x1t-5tyl3** [comment]
I wanna attach #3156 #3160 #2600 to this as they're sort-of related.

**Comment by shiftyscales** [comment]
@3x1t-5tyl3 I disagree that the above are related, and having to consider those issues alongside this one would make it less likely we'd prioritize it.

@amplified1 As you highlighted we likely would not make a physical locomotion into noclip. Can you clarify what / how you are wanting to modify the noclip locomotion in specific / what you are trying to accomplish? There may be better ways to handle it, e.g. adding additional settings like noclip speed.

**Comment by amplified1** [comment]
The noclip locomotion can be used as a basis for highly specific custom locomotions made with protoflux because it has no built in gravity or collisions. Settings related to the normal functionality of noclip would be irrelevant in this case, and it is also reasonable that if a more specific experience/game world wanted to have fine-tuned control over noclip properties and speed, that they would want to be able to set it themselves, not leave it up to a local user setting

**Comment by shiftyscales** [comment]
In the context of a game, I would recommend relevant colliders are set up with a local state that toggles player collisions- this allows you to let the player move through walls without moving through the floor / ceiling / out of the playable bounds.

Noclip is intended for use as a development locomotion, and is also not suitable as it could not clip with any triggers / other collider types you set up.

**Comment by ko-tengu** [comment]
A real world example that I've run into is that I have an avatar that has toggles to be a ghost, and as a fun Resonite gimmick I wanted to set it up to automatically go to No Clip locomotion (if available) when I turned on the ghost toggles. With the current limitations I can only do this by having a second No Clip Locomotion option in my menu and using protoflux to switch to this specific version that lives inside my avatar and make sure it's installed on avatar equip. Either #3160 or being able to rename the locomotion module or preferably both would help streamline this and make it less messy.

**Comment by amplified1** [comment]
Locally disabled collisions cannot work with arbitrary environments, it's also worth bringing up that the noclip locomotion has built in variance. You can have a 6dof rotation noclip module, this is built-in functionality, and it will be indistinguishable from a normal one. If the noclip locomotion is not worth improved functionality, then locomotion as a whole should be more generic than it currently is. Resonite culture is highly based on avatar functionality independent of the world, and there have been many cases where I have wanted to program custom character controller behavior from scratch because the behavior of the existing modules were completely incapable of fulfilling my needs.
